Sources/LiveKit/Track/AudioManager.swift

```diff
@@ -59,6 +59,29 @@ public class LKAudioBuffer: NSObject {
 
 // Audio Session Configuration related
 public class AudioManager: Loggable {
+    #if os(iOS)
+    class AudioSessionDelegateObserver: NSObject, Loggable, LKRTCAudioSessionDelegate {
+        func audioSessionDidStartPlayOrRecord(_: LKRTCAudioSession) {
+            log()
+        }
+
+        func audioSession(_: LKRTCAudioSession, audioEngineWillUpdateStateWithOutputEnabled isOutputEnabled: Bool, inputEnabled isInputEnabled: Bool) {
+            log("isOutputEnabled: \(isOutputEnabled), isInputEnabled: \(isInputEnabled)")
+
+            // Configure audio session
+            let config = LKRTCAudioSessionConfiguration.webRTC()
+            config.category = AVAudioSession.Category.playAndRecord.rawValue
+            config.categoryOptions = [.allowBluetooth, .allowBluetoothA2DP, .allowAirPlay]
+            config.mode = AVAudioSession.Mode.videoChat.rawValue
```

Comment:
my 2c: it might be worth allowing end-users to customize these options. our current setup *requires* `.defaultToSpeaker`. We also did not allow airplay specifically because of issues with noise cancellation.

(THANK YOU for going down the AVAudioEngine rabbit hole. I know it's a huge one)

Reply:
Yes, I will allow devs to customize the session. 🙂